tezos/new_context/src/working_tree/storage.rs

```diff
@@ -251,11 +254,33 @@ pub struct StorageMemoryUsage {
     strings: StringsMemoryUsage,
 }
 
+/// `Storage` contains all the data from the working tree.
+///
+/// This is where all trees/blobs/strings are allocated.
+/// The working tree only has access to ids which refer to data inside `Storage`.
+///
+/// Because `Storage` is for the working tree only, it is cleared before
+/// every checkout.
```

Comment:
Excellent, this along the comments for each field make it much easier to understand👍